### ISSUE TITLE
How many iterations......

### ISSUE THREAD

**Issue by krishnak** [Linzaer/Ultra-Light-Fast-Generic-Face-Detector-1MB#167]
@Linzaer 

I would like to know during your training of the model,how many iterations were completed and what was the error at the end of the iterations

**Comment by Linzaer** [comment]
You can choose the model with the minimum validation loss. If you use SGD and default configuration,it's about 150 epochs. There is no error at the end of the iterations.

**Comment by krishnak** [comment]
thank you, I have been using Adam and a different loss function, I have done about 60 iterations with the 16000 image dataset, my losses are some where at 14. However the detector is detecting as below. I want to know whether it will improve if I run to 150 iterations or should I change the loss function.
![Figure_1](https://user-images.githubusercontent.com/1259447/76917527-b1ed2980-68e9-11ea-8eba-caf8517ace9b.png)
![Figure_4](https://user-images.githubusercontent.com/1259447/76917743-5b341f80-68ea-11ea-8182-1cd6d37b677a.png)

**Comment by Linzaer** [comment]
I don't think it will improve because Adam converges quickly. Are boxes normal when you use the default loss function ?

**Comment by krishnak** [comment]
I haven't tested your default loss function as I have changed a few things to remove the softmax operation, my loss function is similar to [this](https://gist.github.com/krishnak/a869ca18e5b14564bc01dbf37cba8d28)

**Comment by krishnak** [comment]
My issue is not about bounding box errors, that may have some bug in my code, but in detection of faces itself. As you can see the not all faces are being detected, I am having a detection threshold of 0.3 during training. The above picture is infered with a detection threshold of 0.5

**Comment by krishnak** [comment]
@Linzaer After several days I figured out the reason for the losses not dropping , I use image augmentation with your training set, that seems to hold the losses as it is.

I removed the image augmentation, however I get NAN error if I run the full dataset without augmentation. If I use (randomly) 5000 files for training and 4000 for validation, NAN issue goes away, do you have any idea why this occurs. If I use image augmentation on 16170 training set of yours, I never get NAN

**Comment by krishnak** [comment]
@Linzaer Just a follow up, did you use image augmentation during your training? It appears that if I don't use image augmentation, I can drive error down to zero - I tried it with a small set of images.

However when I use the whole image set without image augmentation, I get NAN error and it doesn't complete an epoch. It kind of hits NAN after several thousand images in the first epoch (it is mostly during validation run)

If I pick a training size of less than 10K images (randomly) then the training goes through without NAN error. Can you think of what could be the cause for this.

**Comment by krishnak** [comment]
I have fixed most of the issues, for any one having similar issues. If you use image augmentation, it will take a longer time for the training to complete considering the dataset is large. So augmentation can be turned off. Secondly NAN issues crop up if you have high resolution images when they are scaled down, the bounding boxes become close to 0. Most importantly if you set your learning rate to very low value it takes a long time
